developer-docs/latest/concepts/sso.md

```diff
@@ -0,0 +1,320 @@
+          uid: 'github',
+          displayName: 'Github',
+          icon: 'https://cdn1.iconfinder.com/data/icons/logotypes/32/github-512.png',
+          createStrategy: strapi => new GithubStrategy({
+            clientID: env('GITHUB_CLIENT_ID'),
+            clientSecret: env('GITHUB_CLIENT_SECRET'),
+            scope: ['user:email'],
+            callbackURL: strapi.admin.services.passport.getStrategyCallbackURL('github'),
+          }, (accessToken, refreshToken, profile, done) => {
+            done(null, {
+              email: profile.emails[0].value,
+              username: profile.username,
+            });
+          }),
+        },
+      ],
+    },
+  },
+});
+```
+:::
+
+::: tab Discord
+
+Using: [passport-discord](https://github.com/nicholastay/passport-discord#readme)
+
+```bash
+yarn add passport-discord
+```
+
+`/config/server.js`
+
+```jsx
+"use strict";
+
+const DiscordStrategy = require("passport-discord");
+
+module.exports = ({ env }) => ({
+  // ...
+  admin: {
+    // ...
+    auth: {
+      // ...
+      providers: [
+        {
+          uid: 'discord',
+          displayName: 'Discord',
+          icon: 'https://cdn0.iconfinder.com/data/icons/free-social-media-set/24/discord-512.png',
+          createStrategy: strapi =>
+            new DiscordStrategy(
+              {
+                clientID: env('DISCORD_CLIENT_ID'),
+                clientSecret: env('DISCORD_SECRET'),
+                callbackURL: strapi.admin.services.passport.getStrategyCallbackURL(
+                  'discord'
+                ),
+                scope: ['identify', 'email'],
+              },
+              (accessToken, refreshToken, profile, done) => {
+                done(null, {
+                  email: profile.email,
+                  username: `${profile.username}#${profile.discriminator}`,
+                });
+              }
+            ),
+        },
+      ],
+    },
+  },
+});
+```
+:::
+::::
+
+## Advanced Customization
+
+### Admin Panel URL
+
+If your administration panel lives on a different host/port than your Strapi server, you will need to modify the admin URL.
+To do so, head to your `/config/server.js` configuration file and tweak the `admin.url` field.
+
+For example, if your admin application has been started on `localhost:4000`, your configuration will look like the following:
+
+```javascript
+{
+  ...
+  admin: {
+    url: 'http://localhost:4000/admin',
+    ...
+  }
+}
+```
+
+### Custom Logic
+
+In some scenarios, you will want to write additional logic for your connection workflow such as:
+- Restricting connection and registration for a specific domain
+- Triggering actions on connection attempt
+- Analytics
+
+The easiest way to do so is to plug into the verify function of your strategy and write some code.
+
+For example, if you want to allow only people with an official strapi.io email address, you can instantiate your strategy like this:
+
+```javascript
+const strategyInstance = new Strategy(
```

Comment:
Need location of where this would be, if it's the server.js we should clarify that, at the moment it's too abstract. Does this go in `./config/functions`?

Reply:
The point here was not to showcase a file but a way to structure and build your strategies. (which you then use in your server.js config file)